chia/full_node/full_node.py

```diff
@@ -709,101 +709,96 @@ async def _sync(self):
-        advanced_peak = False
+        peers_with_peak = self.get_peers_with_peak(peak_hash)
+        fork_point_height = await check_fork_next_block(
+            self.blockchain, fork_point_height, peers_with_peak, node_next_block_check
+        )
         batch_size = self.constants.MAX_BLOCK_COUNT_PER_REQUESTS
 
-        our_peak_height = self.blockchain.get_peak_height()
-        ses_heigths = self.blockchain.get_ses_heights()
-        if len(ses_heigths) > 2 and our_peak_height is not None:
-            ses_heigths.sort()
-            max_fork_ses_height = ses_heigths[-3]
-            # This is the fork point in SES in the case where no fork was detected
-            if self.blockchain.get_peak_height() is not None and fork_point_height == max_fork_ses_height:
-                for peer in peers_with_peak:
-                    # Grab a block at peak + 1 and check if fork point is actually our current height
-                    block_response: Optional[Any] = await peer.request_block(
-                        full_node_protocol.RequestBlock(uint32(our_peak_height + 1), True)
-                    )
-                    if block_response is not None and isinstance(block_response, full_node_protocol.RespondBlock):
-                        peak = self.blockchain.get_peak()
-                        if peak is not None and block_response.block.prev_header_hash == peak.header_hash:
-                            fork_point_height = our_peak_height
-                        break
-
-        for i in range(fork_point_height, target_peak_sb_height, batch_size):
-            start_height = i
-            end_height = min(target_peak_sb_height, start_height + batch_size)
-            request = RequestBlocks(uint32(start_height), uint32(end_height), True)
-            self.log.info(f"Requesting blocks: {start_height} to {end_height}")
-            batch_added = False
-            to_remove = []
-            for peer in peers_with_peak:
-                if peer.closed:
-                    to_remove.append(peer)
-                    continue
-                response = await peer.request_blocks(request, timeout=60)
-                if response is None:
-                    await peer.close()
-                    to_remove.append(peer)
-                    continue
-                if isinstance(response, RejectBlocks):
-                    to_remove.append(peer)
-                    continue
-                elif isinstance(response, RespondBlocks):
-                    success, advanced_peak, _ = await self.receive_block_batch(
-                        response.blocks, peer, None if advanced_peak else uint32(fork_point_height), summaries
-                    )
-                    if success is False:
-                        await peer.close(600)
+        async def fetch_block_batches(batch_queue, peers_with_peak: List):
+            for start_height in range(fork_point_height, target_peak_sb_height, batch_size):
```

Comment:
Let's wrap inside of `fetch_block_batches` in try/finally and add `None` to the `batch_queue` in the `finally` statement, just in case there is exception being thrown somewhere in there, we don't want to perma-block the validation.

Reply:
good idea

Reply:
ok i added that, but its a bit strange as this is wrapped in a try/finally in sync_from_fork_point although i guess we don't want to exit the entire sync flow if we have some exception